pkg/cmd/adm/install_operator.go

```diff
@@ -0,0 +1,236 @@
+						Duration: 5 * time.Minute,
+					},
+				},
+			},
+		},
+	}
+}
+
+func newOperatorGroup(name types.NamespacedName) *olmv1.OperatorGroup {
+	return &olmv1.OperatorGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: olmv1.OperatorGroupSpec{
+			TargetNamespaces: []string{
+				name.Namespace,
+			},
+		},
+	}
+}
+
+func newSubscription(name types.NamespacedName, operatorName, catalogSourceName string) *olmv1alpha1.Subscription {
+	return &olmv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: name.Namespace,
+			Name:      name.Name,
+		},
+		Spec: &olmv1alpha1.SubscriptionSpec{
+			Channel:                "staging",
+			InstallPlanApproval:    olmv1alpha1.ApprovalAutomatic,
+			Package:                operatorName,
+			CatalogSource:          catalogSourceName,
+			CatalogSourceNamespace: name.Namespace,
+		},
+	}
+}
+
+func waitUntilCatalogSourceIsReady(ctx *clicontext.TerminalContext, catalogSourceKey runtimeclient.ObjectKey, waitForReadyTimeout time.Duration) error {
+	cs := &olmv1alpha1.CatalogSource{}
+	if err := wait.PollImmediate(2*time.Second, waitForReadyTimeout, func() (bool, error) {
+		ctx.Printlnf("waiting for CatalogSource %s to become ready", catalogSourceKey)
+		if err := ctx.KubeClient.Get(ctx, catalogSourceKey, cs); err != nil {
+			return false, err
+		}
+
+		return cs.Status.GRPCConnectionState != nil && cs.Status.GRPCConnectionState.LastObservedState == "READY", nil
+	}); err != nil {
+		csString, _ := json.Marshal(cs)
+		return errs.Wrapf(err, "failed waiting for catalog source to be ready.\n CatalogSrouce found: %v \n\t", string(csString))
+	}
+	return nil
+}
+
+func waitUntilInstallPlanIsComplete(ctx *clicontext.TerminalContext, cl runtimeclient.Client, namespace string, waitForReadyTimeout time.Duration) error {
+	plans := &olmv1alpha1.InstallPlanList{}
+	if err := wait.PollImmediate(2*time.Second, waitForReadyTimeout, func() (bool, error) {
+		ctx.Printlnf("waiting for InstallPlans in namespace %s to complete", namespace)
+		if err := cl.List(ctx, plans, runtimeclient.InNamespace(namespace),
+			runtimeclient.MatchingLabels{fmt.Sprintf("operators.coreos.com/%s.%s", namespace, namespace): ""},
+		); err != nil {
+			return false, err
+		}
+
+		for _, ip := range plans.Items {
+			if ip.Status.Phase != olmv1alpha1.InstallPlanPhaseComplete {
+				return false, nil
+			}
+		}
+
+		return len(plans.Items) > 0, nil
+	}); err != nil {
+		plansString, _ := json.Marshal(plans)
+		return errs.Wrapf(err, "failed waiting for install plan to be complete.\n InstallPlans found: %s \n\t", string(plansString))
+	}
+	return nil
+}
+
+// checkOneOperatorPerNamespace returns an error in case the namespace contains the other operator installed.
+// So for host namespace member operator should not be installed in there and vice-versa.
+func checkOneOperatorPerNamespace(ctx *clicontext.TerminalContext, namespace, operator string) error {
```

Comment:
We probably also shouldn't have 2 members in the same namespace.. Or 2 host operators for that matter...

Reply:
Right now, if you run it twice for the same operator in the same namespace, it should fail when creating the CatalogSource. But we can have a check and make the error more user friendly in case.